src/component/tooltip/TooltipHTMLContent.ts

```diff
@@ -358,6 +358,34 @@ class TooltipHTMLContent {
             + `border-color: ${nearPointColor};`
             + (tooltipModel.get('extraCssText') || '');
 
+        const userWidth = tooltipModel.get(['textStyle', 'width']);
+        if (userWidth != null) {
+            el.style.cssText += `;width:${userWidth}px;`;
+            // `text-overflow_string` has very humble compatibility
+            // shttps://caniuse.com/mdn-css_properties_text-overflow_string
+            const ellipsis = tooltipModel.get(['textStyle', 'ellipsis']);
+            const userOverflow = tooltipModel.get(['textStyle', 'overflow']);
+            switch (userOverflow) {
+                case 'truncate':
+                    el.style.cssText += ';overflow:hidden;'
+                        + `text-overflow:${ellipsis != null ? `\'${ellipsis}\'` : 'ellipsis'};`
+                        + 'white-space:nowrap;';
+                    break;
+
+                case 'break':
+                    el.style.cssText += ';word-break:break-word;white-space:normal;';
+                    break;
+
+                case 'breakAll':
+                    el.style.cssText += ';word-break:break-all;white-space:normal;';
+                    break;
+
+                default:
+                    el.style.cssText += '';
+                    break;
+            }
+        }
```

Comment:
I'm thinking this logic can be simplified a bit. Just FYI.

```js
const ellipsis = retrieve(tooltipModel.get(['textStyle', 'ellipsis']), 'ellipsis');
const userOverflow = tooltipModel.get(['textStyle', 'overflow']);
if (userOverflow) {
    let overflowStyle;
    if (userOverflow === 'truncate') {
        overflowStyle = `overflow:hidden;text-overflow:${ellipsis};white-space:nowrap;`
    }
    else {
        const breakMode = userOverflow === 'break'
            ? 'break-word'
            : userOverflow === 'breakAll'
                ? 'break-all' : '';
        breakMode && (overflowStyle = `word-break:${breakMode};white-space:normal;`);
    }
    el.style.cssText += overflowStyle;
}
```

Reply:
I have reservation on this, `swtich` makes code more readable when multiple conditional branch, but on the other hand `if else` has better performance. @plainheart

Reply:
I can understand it. The reason for the proposal is just for reducing the minimized code size by merging some conditions. Sure, except for that, we can also make the switch simpler.

Reply:
My humble option:

`switch...case...`  and `if...else...` has minor performance in most cases. I think they both OK. Code size and readability are the main concerns in this case. I will prefer @plainheart solution. It reduces much redundant strings and is clear enough from my side.

Reply:
Ok, `if else` it is.